services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/core/BatchBuffer.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.sqs.internal.batchmanager.core;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledFuture;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+
+/**
+ * Interface representing a buffer that handles batching of requests and responses.
+ *
+ * @param <RequestT>  the type of the request
+ * @param <ResponseT> the type of the response
+ */
+@SdkInternalApi
+public interface BatchBuffer<RequestT, ResponseT> {
```

Comment:
Agree, I removed this interface . I created it initially thinking this could be shared with ResponseManager but response batch manager buffer behaviour is different so  we dont need the interface.